src/Compilers/CSharp/Portable/Binder/Semantics/BestTypeInferrer.cs

```diff
@@ -258,7 +248,9 @@ private static TypeSymbol Better(TypeSymbol type1, TypeSymbol type2, Conversions
 
                 if (type1.Equals(type2, TypeCompareKind.IgnoreDynamicAndTupleNames))
                 {
-                    return MethodTypeInferrer.Merge(type1, type2, conversions.CorLibrary);
+                    TypeSymbol merged;
+                    MethodTypeInferrer.TryMerge(type1, type2, conversions.CorLibrary, out merged);
```

Comment:
>TryMerge [](start = 39, length = 8)

Can `TryMerge` return false?

Reply:
It looks like this version of `TryMerge` uses `VarianceKind.None`. I think it should be using the covariant rule, at least at the top level.

---
In reply to: [223450999](https://github.com/dotnet/roslyn/pull/29727#discussion_r223450999) [](ancestors = 223450999)

Reply:
`TryMerge` can return `false`. In all cases, we want to return the merged result, including the `false` case where the result is `null`.

---
In reply to: [223451603](https://github.com/dotnet/roslyn/pull/29727#discussion_r223451603) [](ancestors = 223451603,223450999)